File: concrete/src/main/java/pcb/uwu/unit/quantity/NauticalMileUnit.java
package pcb.uwu.unit.quantity;

import pcb.uwu.units.quantity.LengthUnit;

import java.math.BigDecimal;

import static java.math.MathContext.DECIMAL64;

public class NauticalMileUnit extends LengthUnit {

	private static final BigDecimal METERS_PER_NAUTICAL_MILE = new BigDecimal("1852");

	public static final NauticalMileUnit NAUTICAL_MILE = new NauticalMileUnit();

	private NauticalMileUnit() {
		super("mile", "mile", "miles",
				value -> value.multipliedBy(METERS_PER_NAUTICAL_MILE, DECIMAL64),
				value -> value.dividedBy(METERS_PER_NAUTICAL_MILE, DECIMAL64));
	}
}

File: implementation/src/main/java/pcb/uwu/unit/base/NauticalMileUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.unit.base.NauticalMileUnit.CONVERT

object NauticalMileUnit
    : LengthUnit(symbol = "mile",
                 singularName = "mile",
                 pluralName = "miles",
                 translationToCanonical = { it * CONVERT },
                 translationFromCanonical = { it / CONVERT })
{
    const val CONVERT = 1852
}
